source/behaviors/offClick/offClick.ts

```diff
@@ -29,16 +30,36 @@ export class OffClickDirective implements OnInit, OnDestroy {
 	}
 
 	ngOnInit() {
-		setTimeout(() => {
-			document.addEventListener('click', this.listener);
-		}, 0);
+		if (this.active) {
+			setTimeout(() => {
+				this.addListener();
+			});
+		}
 	}
 
-	ngOnDestroy() {
+	ngOnChanges(changes: SimpleChanges): void {
```

Comment:
👍  I have a feeling we are gonna get some nice usage out if this, cleaner the better!